packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -580,21 +616,69 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
   /// thumbOffsetLocal is a position in the thumb track. Cannot be null.
   double getTrackToScroll(double thumbOffsetLocal) {
     assert(thumbOffsetLocal != null);
-    final double scrollableExtent = _lastMetrics!.maxScrollExtent - _lastMetrics!.minScrollExtent;
+    late double scrollableExtent;
+    // If the ScrollView has a finite extent, and is not being overridden by
+    // infinite scroll, use the known extents.
+    if (_lastMetrics!.maxScrollExtent.isFinite && infiniteBehavior == null) {
+      scrollableExtent = _lastMetrics!.maxScrollExtent - _lastMetrics!.minScrollExtent;
+    } else {
+      // If we have an infinitely long scroll view (or are pretending to have
+      // one), we pretend it isn't by keeping track of the last known depth, and
+      // adding on more 'extent' based on the InfiniteScrollBehavior.
+      scrollableExtent = _infiniteDepth - _lastMetrics!.minScrollExtent;
+    }
     final double thumbMovableExtent = _trackExtent - _thumbExtent();
-
     return scrollableExtent * thumbOffsetLocal / thumbMovableExtent;
   }
 
   // Converts between a scroll position and the corresponding position in the
   // thumb track.
   double _getScrollToTrack(ScrollMetrics metrics, double thumbExtent) {
-    final double scrollableExtent = metrics.maxScrollExtent - metrics.minScrollExtent;
+    late double scrollableExtent;
+    final double trueScrollableExtent = metrics.maxScrollExtent - metrics.minScrollExtent;
+    // If the ScrollView has a finite extent, and is not being overridden by
+    // infinite scroll, use the known extents.
+    if (metrics.maxScrollExtent.isFinite && infiniteBehavior == null) {
```

Comment:
This should be 
`if (metrics.maxScrollExtent.isFinite) {`
right?

Reply:
Well, that's one of the questions. Here, I am allowing developers to opt-in to infinite behavior if they want to, even is their scroll view has a finite length. For folks with dynamically sized items in a lazy-building list, the scrollbar can sometimes skip around as children are built because the maxExtent is estimated in those cases (#25652).

This can be seen in the wild as expected behavior (the twitter clip is a good example), but some folks have expressed a desire to have a smoother behavior. This could be a way to provide that.